libtransmission/log.cc

```diff
@@ -197,12 +197,14 @@ void tr_logFreeQueue(tr_log_message* freeme)
 char* tr_logGetTimeStr(char* buf, size_t buflen)
 {
     auto const a = std::chrono::system_clock::now();
+    auto const a_tm = fmt::localtime(std::chrono::system_clock::to_time_t(a));
     auto const [out, len] = fmt::format_to_n(
         buf,
         buflen - 1,
-        "{0:%F %H:%M:}{1:%S}",
-        a,
-        std::chrono::duration_cast<std::chrono::milliseconds>(a.time_since_epoch()));
+        "{0:%FT%R:}{1:%S}{2:%z}",
```

Comment:
I don't suppose ISO8610 mentions the locale, yet `{:%S}` [docs](https://fmt.dev/11.0/syntax/#chrono-format-specifications) say that "The character for the decimal point is localized according to the locale." Might be better to print the milliseconds separately/explicitly (i.e. print `{:%S}` from `tm`, and `{:%03}` for milliseconds part), so that it's also present regardless of whether it can "be exactly represented with seconds" (same docs).

Reply:
Both `.` and `,` are valid decimal points according to ISO8601's Wiki page.

Reply:
Those aren't the only possible separators though, are they? Based on https://en.wikipedia.org/wiki/Decimal_separator#Examples_of_use, some countries _used to_ use "·"… Let's just hope they won't switch back in the near future :)

Any opinion on ".000" being dropped?

Reply:
`.000` won't be dropped, because the precision of the milliseconds format is 3 decimal places, see https://godbolt.org/z/Mz174Eqfa.

As for the decimal point character, the docs says it is locale dependent, but the code says otherwise... I've raised an issue upstream https://github.com/fmtlib/fmt/issues/4117.

Reply:
e0e7988d3ffea5d5e0b8652fff2cf9cb8cb1f495 Anyway, this thread took some detours because of me, but it does seem like hard coding the decimal point ourselves is a better option.

Upon looking further into this issue, seems like the C++ standard is still bit of a mess in this regard.